path_credentials.go

```diff
@@ -301,6 +319,70 @@ func saveToken(ctx context.Context, token *astraToken, s logical.Storage) error
 	return nil
 }
 
+func doesTokensMatch(token *astraToken, tokentoRevoke string ) bool {
+	return token.Token == tokentoRevoke
+}
+
+func (b *datastaxAstraBackend) tokenRevoke(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	tokenRaw, ok := req.Secret.InternalData["token"]
+	if !ok {
+		return nil, errors.New("token not retrived")
+	}
+	
+	tokens, err := listCreds(ctx, req.Storage)
+	if err != nil {
+		return nil, errors.New("no tokens found")
+	}
+
+	for i := 0; i < len(tokens); i++ {
+		token, err := readToken(ctx, req.Storage, tokens[i])
+		if err != nil {
+			return nil, errors.New("no tokens found")
+		}
+		if doesTokensMatch(token, tokenRaw.(string)) {
+			err = req.Storage.Delete(ctx, "token/"+tokens[i])
+			if err != nil {
+				return nil, err
+			}
+			conf, err := getConfig(ctx, req.Storage, token.OrgID)
+			if err != nil {
+				return nil, err
+			}
+			if conf.URL != "" {
+				client := &http.Client{}
+				url := conf.URL + "/v2/clientIdSecrets/" + token.ClientID
+				httpReq, err := http.NewRequest(http.MethodDelete, url, nil)
+				if err != nil {
+					msg := "error creating httpReq " + err.Error()
+					return nil, errors.New(msg)
+				}
+				httpReq.Header.Add("Content-Type", "application/json")
+				httpReq.Header.Add("Authorization", "Bearer "+conf.AstraToken)
+				res, err := client.Do(httpReq)
+				if err != nil {
+					msg := "error sending request " + err.Error()
+					return nil, errors.New(msg)
+				}
+				defer res.Body.Close()
+				if res.StatusCode != http.StatusNoContent {
+					return nil, errors.New("could not delete token in astra")
+				}
+			} else {
+				return nil, errors.New("config not found")
+			}
+		}
+	}
+	b.reset()
+	return nil, nil
+}
+
+func (b *datastaxAstraBackend) tokenRenew(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	resp := &logical.Response{Secret: req.Secret}
+	// resp.Secret.TTL = 30 * time.Second
+	return resp, nil
+}
```

Comment:
This doesn't seem complete. 

I tried to 

`vault write astra/org/token org_id="org_id" role_name="Organization_Administrator" logical_name="org_admin_token" lease_time=60s`

And then 
`vault lease renew astra/org/token/ZBwNFoMUolS6fxbZnq2RXM0x`

 But the renew did not work.